In [2]:
#Versão da Linguagem Python
from platform import python_version
print('A Versão da Linguagem Python Usada neste Jupyter Notebook:', python_version())

A Versão da Linguagem Python Usada neste Jupyter Notebook: 3.12.3


In [3]:
# realizando os imports necessarios
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt

# <b>Carregando os Dados</b>


In [5]:
# Carrega o dateset
df_dsa = pd.read_csv('dataset.csv')

In [7]:
# Shape - mostra a quantidade de linhas e colunas do dataset
df_dsa.shape

(9700, 11)

In [8]:
# Mostra as 5 primeiras linhas do dataset
df_dsa.head()

ID_Pedido Data_Pedido ID_Cliente   Segmento           Pais  \
0  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
1  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
2  CA-2017-138688  12/06/2017   DV-13045  Corporate  United States   
3  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   
4  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   

            Cidade      Estado       ID_Produto        Categoria SubCategoria  \
0        Henderson    Kentucky  FUR-BO-10001798        Furniture    Bookcases   
1        Henderson    Kentucky  FUR-CH-10000454        Furniture       Chairs   
2      Los Angeles  California  OFF-LA-10000240  Office Supplies       Labels   
3  Fort Lauderdale     Florida  FUR-TA-10000577        Furniture       Tables   
4  Fort Lauderdale     Florida  OFF-ST-10000760  Office Supplies      Storage   

   Valor_Venda  
0     261.9600  
1     731.9400  
2      14.6200  
3     957.5775  
4      22.3680

In [9]:
# Mostra as 5 ultimas linhas do dataset
df_dsa.tail()

ID_Pedido Data_Pedido ID_Cliente     Segmento           Pais  \
9695  CA-2018-154116  15/12/2018   KM-16660     Consumer  United States   
9696  CA-2018-154116  15/12/2018   KM-16660     Consumer  United States   
9697  CA-2018-154116  15/12/2018   KM-16660     Consumer  United States   
9698  CA-2017-105291  30/10/2017   SP-20920     Consumer  United States   
9699  CA-2018-147032  31/07/2018   LB-16795  Home Office  United States   

               Cidade      Estado       ID_Produto        Categoria  \
9695        Inglewood  California  OFF-PA-10004569  Office Supplies   
9696        Inglewood  California  OFF-AP-10000027  Office Supplies   
9697        Inglewood  California  TEC-PH-10000675       Technology   
9698  San Luis Obispo  California  OFF-FA-10003059  Office Supplies   
9699       Wilmington    Delaware  OFF-PA-10003256  Office Supplies   

     SubCategoria  Valor_Venda  
9695        Paper       22.830  
9696   Appliances       54.320  
9697       Phones      196.776  
9698    Fasteners        3.620  
9699        Paper       11.540

# <b> Análise Exploratória </b>


In [11]:
# Mostra os tipos de colunas do conjunto de dados
df_dsa.columns

Index(['ID_Pedido', 'Data_Pedido', 'ID_Cliente', 'Segmento', 'Pais', 'Cidade',
       'Estado', 'ID_Produto', 'Categoria', 'SubCategoria', 'Valor_Venda'],
      dtype='object')

In [13]:
# Verifica o tipo de dado de cada coluna no dataset
df_dsa.dtypes

ID_Pedido        object
Data_Pedido      object
ID_Cliente       object
Segmento         object
Pais             object
Cidade           object
Estado           object
ID_Produto       object
Categoria        object
SubCategoria     object
Valor_Venda     float64
dtype: object

In [14]:
# Resumo estatístico da coluna com o valor de venda
df_dsa['Valor_Venda'].describe()

count     9700.000000
mean       230.469892
std        627.504252
min          0.444000
25%         17.248000
50%         54.272000
75%        209.932500
max      22638.480000
Name: Valor_Venda, dtype: float64

In [16]:
# Verifica se a valores duplicados nas colunas
df_dsa[df_dsa.duplicated()]

Empty DataFrame
Columns: [ID_Pedido, Data_Pedido, ID_Cliente, Segmento, Pais, Cidade, Estado, ID_Produto, Categoria, SubCategoria, Valor_Venda]
Index: []

In [17]:
# Verificando valores ausentes
df_dsa.isnull().sum()

ID_Pedido       0
Data_Pedido     0
ID_Cliente      0
Segmento        0
Pais            0
Cidade          0
Estado          0
ID_Produto      0
Categoria       0
SubCategoria    0
Valor_Venda     0
dtype: int64

# <b>Pergunta de Negócio 1:</b>
# Qual cidade com maior valor de venda de produtos da categoria 'Office Supplies'?

# <b>Pergunta de Negócio 2:</b>
# Qual o total de vendas por data do pedido?
Demonstre o resultado através de um gráfico de barras

# <b>Pergunta de Negócio 3:</b>
# Qual o total de vendas por Estado?
Demonstre o resultado através de um gráfico de barras

# <b>Pergunta de Negócio 4:</b>
# Quais são as 10 cidades com maior total de vendas?
Demonstre o resultado através de um gráfico de barras


# <b>Pergunta de Negócio 5:</b>
# Qual o segmento teve o maior total de vendas?
Demonstre o resultado através de um gráfico de pizza

# <b>Pergunta de Negócio 6 (Desafio Nível Baby):</b>
# Qual o de vendas por segmento e por ano?

# <b>Pergunta de Negócio 7 (Desafio Nível Júnior):</b>
Os gestores da empresa estão considerando conceder diferentes faixas de descontos e gostariam de fazer uma simulação com base na regra abaixo:
* Se o Valor_Venda for maior que 1000 recebe 15% de desconto
* Se o Valor_Venda for menor que 1000 recebe 10% de desconto
# Quantas vendas receberiam 15% de desconto?

# <b>Pergunta de Negócio 8 ( Desasfio Nível Master):</b>
# Considere que a empresa decida conceder o desconto de 15% do item anterior. Qual seria a média do valor de venda antes e depois do desconto?

# <b>Pergunta de Negócio 9 (Desafio Nível Master Ninja):</b>
# Qual a média de vendas por segmento, por ano e por mês?

# <b>Pergunta de Negócio 10 ( Desafio Nível Master Ninja das Galáxias):</b>
# Qual o total de vendas por categoria e subcategoria, considerando somente as TOP 12 subcategorias?
Demonstre tudo através de um único gráfico 